L1Trigger/L1TTwinMux/src/DTRPCBxCorrection.cc

```diff
@@ -44,122 +45,140 @@ void DTRPCBxCorrection::run( const edm::EventSetup& c) {
+ // L1MuDTChambPhDigi * rpcts1=0;
+  auto dtts=std::make_shared<L1MuDTChambPhDigi>(0,0,0,0,0,0,7,0,0,0);
+  auto dtts_sh2nd=std::make_shared<L1MuDTChambPhDigi>(0,0,0,0,0,0,7,0,0,0);
+  auto dttsnew=std::make_shared<L1MuDTChambPhDigi>(0,0,0,0,0,0,7,0,0,0);
+  auto rpcts1=std::make_shared<L1MuDTChambPhDigi>(0,0,0,0,0,0,7,0,0,0);
+  std::auto_ptr<L1MuTMChambPhContainer> m_phiDTDigis_tm(new L1MuTMChambPhContainer);
+  const std::vector<L1MuDTChambPhDigi> *phiChambVectorDT;
+  phiChambVectorDT= m_phiDTDigis.getContainer();
+  m_phiDTDigis_tm->setContainer(*phiChambVectorDT);
+  std::auto_ptr<L1MuTMChambPhContainer> m_phiRPCDigis_tm(new L1MuTMChambPhContainer);
+  const std::vector<L1MuDTChambPhDigi> *phiChambVectorRPC;
+  phiChambVectorRPC= m_phiRPCDigis.getContainer();
+  m_phiRPCDigis_tm->setContainer(*phiChambVectorRPC);
 
   int bx=0, wheel=0, sector=0, station=1;
   int ibx_dtm = 0, fbx_dtm = 0;
   int ibx_dtp = 0, fbx_dtp = 0;
 
-for (wheel=-2;wheel<=2; wheel++ ){
-    for (sector=0;sector<12; sector++ ){
+  for (wheel=-2;wheel<=2; wheel++ ){
+   for (sector=0;sector<12; sector++ ){
      for (station=1; station<=4; station++){
-    	bool shifted[7] = {false, false, false, false,false, false, false};
-    	bool dups[7] = {false, false, false, false,false, false, false};
-			bool secondTs[7] = {false, false, false, false,false, false, false};
-
-    	L1MuTMChambPhContainer shiftedPhiDTDigis;
-			L1MuDTChambPhDigi *dtts_sh2nd = 0;
-
-        for(bx=3; bx>=-3; bx--){
-        vector<int> delta_m, delta_p, delta_0;
-
-          for(int rpcbx=bx-1; rpcbx<=bx+1; rpcbx++){
-            dtts=0; rpcts1=0; dttsnew = 0;
-            dtts = m_phiDTDigis.chPhiSegm(wheel,station,sector,bx ,track_seg);
-            if(!dtts) continue;
-
-            int nhits = noRPCHits(m_phiRPCDigis, rpcbx, wheel, sector, station);
-            for(int hit=0; hit<nhits; hit++){
-            rpcts1 = m_phiRPCDigis.chPhiSegm(wheel, station, sector, rpcbx,hit);
-						///Store in vectors the dphi of matched dt/rpc
-            if(rpcts1 && dtts && dtts->code()<m_QualityLimit && deltaPhi(dtts->phi(),rpcts1->phi()) < m_DphiWindow){
-		           if((dtts->bxNum()-rpcbx)==-1  ) {
-		             delta_m.push_back( deltaPhi(dtts->phi(),rpcts1->phi()) );
-                 ibx_dtm = dtts->bxNum(); fbx_dtm = rpcbx;
-               }
-
+       bool shifted[7] = {false, false, false, false,false, false, false};
+       bool dups[7] = {false, false, false, false,false, false, false};
+       bool secondTs[7] = {false, false, false, false,false, false, false};
+       L1MuTMChambPhContainer shiftedPhiDTDigis;
+       //L1MuDTChambPhDigi *dtts_sh2nd = 0;
+       for(bx=3; bx>=-3; bx--){
+         vector<int> delta_m, delta_p, delta_0;
+         for(int rpcbx=bx-1; rpcbx<=bx+1; rpcbx++){
+           //dtts=0; rpcts1=0; dttsnew = 0;
+           //dtts = m_phiDTDigis_tm->chPhiSegm(wheel,station,sector,bx ,track_seg);
+            dtts.reset(new L1MuDTChambPhDigi(0,0,0,0,0,0,7,0,0,0));
```

Comment:
This will probably give a bad performance since new L1MuDTChampPhDigis are constructed all the time.
Better to keep the raw pointers in this case.